### PR TITLE
In Python3 headers should be str and in Python2 they should be bytes.

### PR DESCRIPTION
**Pull request by Natim**
Fixes mozilla-services/kinto#18

### CODE REVIEW

File: cliquet/tests/resource/test_views.py [hunk 1]
```diff
@@ -294,21 +295,27 @@ def test_next_page_url_has_got_port_number_if_different_than_80(self):
         resp = self.app.get(self.collection_url + '?_limit=1',
                             extra_environ={'HTTP_HOST': 'localhost:8000'},
                             headers=self.headers)
-        next_page = resp.headers['Next-Page'].decode('utf8')
+        next_page = resp.headers['Next-Page']
+        if next_page.__class__ is not six.text_type:
```
Comment (leplatrem) [inline cliquet/tests/resource/test_views.py]: Why not using ``isinstance`` ?

File: cliquet/storage/exceptions.py [hunk 2]
```diff
@@ -15,6 +15,9 @@ def __init__(self, original=None, message=None, *args, **kwargs):
                                   original)
         super(BackendError, self).__init__(message, *args, **kwargs)
 
+    def __str__(self):
+        return 'BackendError: %s' % self.original
+
```
Comment (leplatrem) [inline cliquet/storage/exceptions.py]: Looks like this out of scope :)

File: cliquet/resource.py [hunk 3]
```diff
@@ -110,10 +110,12 @@ def collection_get(self):
         records, total_records, next_page = self.get_records()
 
         headers = self.request.response.headers
-        headers['Total-Records'] = ('%s' % total_records).encode('utf-8')
+        headers['Total-Records'] = ('%s' % total_records)
 
         if next_page:
-            headers['Next-Page'] = next_page.encode('utf-8')
+            if next_page.__class__ is not str:
+                next_page = next_page.encode('utf-8')
```
Comment (leplatrem) [inline cliquet/resource.py]: This reminds me:

https://github.com/elastic/elasticsearch-py/issues/113

https://github.com/elastic/elasticsearch-py/commit/18d6b41e8299d3e62bfac2a8c2f8ceb87be24f45

https://github.com/elastic/elasticsearch-py/commit/5fec49e2618ccab706ea360b18d57420478cd265

File: cliquet/tests/resource/test_pagination.py [hunk 4]
```diff
@@ -1,7 +1,8 @@
+import mock
 import random
-from base64 import b64encode, b64decode
+import six
```
Comment (leplatrem) [inline cliquet/tests/resource/test_pagination.py]: six and mock are stdlib ?

File: cliquet/tests/resource/test_views.py (hunk 1 above)
Comment (Natim) [inline cliquet/tests/resource/test_views.py]: Yes I can probably do so, also I reuse the way they are doing it in waitress.

File: cliquet/tests/resource/test_pagination.py (hunk 4 above)
Comment (Natim) [inline cliquet/tests/resource/test_pagination.py]: In python3 mock is, but the difference I do is more Python functionality versus Project specific functionality

**Comment by Natim** [comment]
**Simple is better than complex.**

**Pull request merged**